Task 4 - filtering and searching functionality

mat-kud opened the pull request:
There are problems with grabbing products prices, because of sponsored products

mat-kud committed:
Task 4 - version #1.0

mat-kud committed:
Task 4 - version #1.1 - added .gitignore file

sabinawozniak reviewed (changes requested):
Test verifyTitlesContainChosenBrand failed (maybe only on my computer, because I don't see "Corsair" on the main page): Unable to locate element: {"method":"xpath","selector":"//span[text() = 'Corsair']"}
I think that if you will add action which click "See more", then this test will pass.
And I think that you can use only the second Data Provider and use only this data which you need in the first test (only "Corsair"). It's just a suggestion - your code is ok.

shibiza commented:
Hello! As I can see 

 WebElement brandNameCheckbox = webDriver.findElement(By.xpath("//span[text() = '" + brandName + "']"));

you use triple times in each of your 3 tests, so you may add this variable to your class, not inside @test method

mat-kud committed:
Task 4 - version #1.2 - created private methods for loops, -driver should now expand list with brand names

RoboPen commented:
After asking driver to expand brand list, it doesnt want to iterate over pages

mat-kud committed:
Delete .idea directory

shibiza commented:
You know, the process finished with exit code -1. No one of your @Test works...

mat-kud committed:
Task 4 - version #1.3 - added waiters
- test verifyProductsPricesAreSorted doesnt work proprly due to fact that DOM structure contains "hidden" products

mat-kud committed:
Merge remote-tracking branch 'origin/filtering_and_searching_functionality1' into filtering_and_searching_functionality1

mat-kud commented:
There is a problem with verifyProductsPricesAreSorted test. I found "hidden" products in DOM structure. Thats why test fails

sabinawozniak commented on src/test/java/org/example/FilteringAndSearchingTest.java:
```diff
@@ -0,0 +1,207 @@
+
+        for (int i = 1; i <= numOfPages; i++) {
+//checking if prices are within specified range
+            arePricesInChosenRange = webDriver.findElements(By.xpath("//span[@class=\"a-price\"]/span[@class=\"a-offscreen\"]"))
```
Your xpath finds elements from outside of results list which gives me false negative test result.

![image](https://user-images.githubusercontent.com/100229236/186028464-436dd14f-f368-4a77-ae42-06702dcc2f74.png)

sabinawozniak replied:
Oh sorry, I didn't see your previous comment, that you just found what is wrong ;) I hope that you will deal with this problem.

mat-kud committed:
Task 4 - version #1.4 - updated selector for products prices, improved pages iterator methods

sabinawozniak commented on src/test/java/org/example/FilteringAndSearchingTest.java:
```diff
@@ -0,0 +1,208 @@
+    }
+
+    @Test(dataProvider = "brandNamesAndPriceRanges")
+    public void verifyProductsPricesAreInDefinedRange(String brandName, float minPrice, float maxPrice) {
```
![image](https://user-images.githubusercontent.com/100229236/186164863-16696e16-9d0e-4833-843d-2ec37e5e42a9.png)

![image](https://user-images.githubusercontent.com/100229236/186164741-f937fcab-e46a-4219-bffd-5430f021c939.png)

sabinawozniak commented on src/test/java/org/example/FilteringAndSearchingTest.java:
```diff
@@ -0,0 +1,208 @@
+    }
+
+    @Test(dataProvider = "brandNames")
+    public void verifyProductsPricesAreSortedAscendingly(String brandName) {
```
Same as above test

mat-kud commented:
Tests fail as expected, because conditions are not met. Try to comment after method and check it out by yourself

Petr-Kv commented:
![image](https://user-images.githubusercontent.com/107186284/186173655-17562f68-0a26-49fe-a0cc-069dc7e965eb.png)

Yes, you are right

juliaHrabovska commented on src/test/java/org/example/FilteringAndSearchingTest.java:
```diff
@@ -0,0 +1,208 @@
+        js.executeScript("arguments[0].click();", brandsExpander);
+    }
+
+    /*@AfterTest
```
Commented code shouldn't be in the code you push to a remote repo

juliaHrabovska commented on src/test/java/org/example/FilteringAndSearchingTest.java:
```diff
@@ -0,0 +1,208 @@
+                    .until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//*[contains(@class,'s-pagination-next')]")));
+
+//checking if every title of the page contains brand name
+            everyTitleContainsInputWord = webDriver.findElements(By.xpath("//div[contains(@class,'s-card-container')]//span[contains(@class,'a-size-medium')]"))
```
It is better to make this line shorter. Just move some parts to the next line. The same for other

juliaHrabovska commented on src/test/java/org/example/FilteringAndSearchingTest.java:
```diff
@@ -0,0 +1,208 @@
+        return true;
+    }
+}
+
```
Extra lines

mat-kud committed:
Merge branch 'master' into filtering_and_searching_functionality1
